### DISCUSSION TITLE
Self-Sovereign Identity Plans

### DISCUSSION THREAD

**Discussion by coopbri** [kanidm/kanidm#3215] [Q&A] [answered]
The technology is experimental and very much in draft stages, but does kanidm have plans to eventually support self-sovereign identity (SSI) mechanisms such as decentralized identifiers (DIDs) and verifiable credentials (VCs)?

**Some resources from the space:**
- SSI writeup: https://www.okta.com/identity-101/self-sovereign-identity
- Keycloak OID4VC initial pass PR: https://github.com/keycloak/keycloak/issues/25936
- SpruceKit: https://www.sprucekit.dev
  - Specifically didkit: https://github.com/spruceid/didkit

**Comment by yaleman** [comment]
[DIDs info from W3C](https://www.w3.org/TR/did-core/)

**Comment by Firstyear** [comment]
What is the use case you see here? Kani as the provider of credentials, or a consumer of them?

**Reply by coopbri** [reply]
Both, really, much like implemented in the Keycloak PR (issuer, receiver, verifier). However, this is not really a feature request for kanidm, moreso curious if it might be in the plans (couldn't find info on repo/docs) so I can plan architecture around it since I plan on supporting SSI in my own applications soon!

**Reply by Firstyear** [reply, marked as answer]
We have no plans for it, and we'd like to see more practical users of it before we commit to it.

**Reply by coopbri** [reply]
Got it, no worries! Will use Keycloak instead

EDIT: between this and other missing use cases, I think the README is a bit disingenuous given that this (DIDs/VCs) and e.g. the federated OIDC gateway discussed in https://github.com/kanidm/kanidm/discussions/3163 are already features of competitors (e.g. Keycloak). From README: "You should not need any other components (like Keycloak) when you use Kanidm - **we already have everything you need!**" specifically the bolded section -- that is simply not the case. The first part (keyword *should*) is completely fair, I think that should extend to the rest of that block of the README. More accurate revision example/suggestion:

> You should not need any other components (like Keycloak) when you use Kanidm - we should have everything you need! If not, please check for the existing issues tracking missing features or submit a feature request.

**Reply by Firstyear** [reply]
Yes, but without practical client use cases, we haven't seen a need to implement these specifications. If those use cases existed, we'd be much wore willing to investigate and consider them, but there is not a lot of value to us to spend our time implementing something that won't be used.

**Reply by coopbri** [reply]
That is absolutely no problem! I completely understand that component, I just think it would be fair to avoid making the claim that every possible need is not covered with kanidm, or at least fair to remove the competitor (Keycloak) from the claim. Just feedback as an end user, act on it or don't, it is your project!
